designs/2019-esm-compatibilty/README.md

```diff
@@ -77,6 +77,20 @@ If a user:
 
 They should give the configuration a `.cjs` extension.
 
+### Priority
+
+With the addition of `.cjs`, the new priority for configuration files will be
+
+1. .eslintrc.cjs
+2. .eslintrc.js
+3. .eslintrc.yaml
+4. .eslintrc.yml
+5. .eslintrc.json
+6. .eslintrc
+7. package.json
+
+`.cjs` is placed higher than `.js` because -- unlike the latter -- `.cjs` does not allow for ambiguity. A `.cjs` file is always interpreted as CommonJS regardless of context.
```

Comment:
to be clear, `.js` is always and by default unambiguously CJS, unless in a package.json boundary with `"type": "module"`, when it's unambiguously ESM.

Reply:
> to be clear, .js is always and by default unambiguously CJS, unless in a package.json boundary with "type": "module", when it's unambiguously ESM.

I think that's adding an assumption to the sentence it didn't contain before: That we know anything but "this one file has the extension `.cjs`". With only that information, it is ambiguous in exactly the way you describe. :)

Reply:
(It may be confusing that in other contexts we have used "ambiguous" to describe the problem of files being ambiguous even given all metadata we could possibly look at.)

Reply:
Removed explanation in Revision 6

Reply:
@jkrems a node tool always has the additional default context "based on node's resolution algorithm" :-)